dvc_azure/spec.py

```diff
@@ -0,0 +1,13 @@
+from adlfs import AzureBlobFileSystem as _AzureBlobFileSystem
+
+
+class AzureBlobFileSystem(  # pylint: disable=abstract-method
+    _AzureBlobFileSystem
+):
+    async def _put_file(self, *args, **kwargs) -> None:
+        kwargs["overwrite"] = True
```

Comment:
To be honest, I am curious if removing this would break anything. `put_file` should not overwrite, and just fail if file already exists. 😅

Reply:
Yeah I don't think it's actually needed either since in DVC we generally always are checking exists() beforehand. But I'm also not sure why the default in adlfs is even `overwrite=False`, since every other fsspec filesystem has put_file succeed (and overwrite) even if the destination file already exists. (I guess it's probably a dask requirement?)

Reply:
The overwrite is needed for pushing new versions in cloud versioning.

I am not sure why yet, but the current wrapping (in this P.R) doesn't work as expected whereas the previous does